src/cowboy_webtransport.erl

```diff
@@ -0,0 +1,266 @@
+%% Copyright (c) Loïc Hoguin <essen@ninenines.eu>
+%%
+%% Permission to use, copy, modify, and/or distribute this software for any
+%% purpose with or without fee is hereby granted, provided that the above
+%% copyright notice and this permission notice appear in all copies.
+%%
+%% THE SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES
+%% WITH REGARD TO THIS SOFTWARE INCLUDING ALL IMPLIED WARRANTIES OF
+%% MERCHANTABILITY AND FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR
+%% ANY SPECIAL, DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+%% WHATSOEVER RESULTING FROM LOSS OF USE, DATA OR PROFITS, WHETHER IN AN
+%% ACTION OF CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+%% OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+
+%% @todo To enable WebTransport the following options need to be set:
+%%
+%% QUIC:
+%%  - max_datagram_frame_size > 0
+%%
+%% HTTP/3:
+%%  - SETTINGS_H3_DATAGRAM = 1
+%%  - SETTINGS_ENABLE_CONNECT_PROTOCOL = 1
+%%  - SETTINGS_WEBTRANSPORT_MAX_SESSIONS >= 1
+
+%% Cowboy supports versions 7 through 12 of the WebTransport drafts.
```

Comment:
Actually version 2 should be mostly fine. Maybe note that we send the 02 setting.